designdocs/permissions.md

```diff
@@ -0,0 +1,87 @@
+
+### Details
+From the developer’s perspective it would work like this.
+
+The developer specifies the total permissions they are likely to need during the entire run of the application when requesting the XR Session, using an array with pre-defined permissions constants.  Something like this:
+
+``` javascript
+const perms = [
+   MICROPHONE,
+	 IMMERSIVE_MODE,
+   HEADSET_POSE,
+   //CAMERA_DATA,  developer decides not to use camera
+]
+navigator.xr.requestSession(perms).then(session => {
+	//start up the rest of the app
+})
+```
+
+At this point the user agent can either prompt the user for these permissions, or merely use them as an upper bound of permissions that could potentially be requested later.
+
+
+Later, when the application needs to actually use a permission, say the microphone, the developer will request this access through a promise based API. Ex:
+
+```javascript
+navigator.mediaDevices.getUserMedia(opts).then(stream => {
+	console.log("i have the microphone stream")
+})
+```
+
+At this point the user agent can either prompt the user for the microphone permission, or immediately grant it if the user already approved it previously.  
+
+Note that if the application did not include MICROPHONE as  the initial list of permissions to `requestSession` then the user agent *must* reject any later requests to `getUserMedia()`.  
+
+*The initial list of permissions is the maximum set of permissions the application may request throughout it’s lifetime.*  This enables, say, a vr chat client to let the user enter a move and talk only mode and later request camera access because the user wants to share something, without having to bug the user initially; if the user agent supports immersive permission requests.  On user agents which do not support immersive permission requests, for whatever reason (form factor, user prefs, domain name, phase of moon, etc.), then the request for camera can be done at application start.
+
+
+### Advantages
+This system has two advantages over other approaches.
+* the developer has a single code path that will always be followed.  They will not need to write conditional code to work around different user agent authentication systems.
+* this gives the user agent maximum flexibility to adapt behavior to the current situation. 
+
+User agents could implement many different behaviors which would be perfectly valid under this spec.
+
+* request all permissions of the user at start, as a series of prompts
+* collapse all permissions into a single prompt
+* auto-approve certain permissions based on domain (ex: anything from mycompany.com is always approved)
+* auto-approve certain permissions when others are requested. (ex: if dev asks for camera, we might as well give them microphone too)
```

Comment:
I'd suggest this line be removed, it _might_ be in conflict with other (non WebXR) specification requirements for user consent. For example, does this line say that if another specification requires user consent, the use of WebXR overrides that requirement?

Reply:
Maybe instead we add something like "where allowable by related specifications"?

Reply:
This section is just a list of potential options for how the user agent can interpret the request. It does not require anything. It's entirely up to the user agent.

Reply:
I added wording to clarify that this section is non-prescriptive and UAs are still bound by other related specifications.